File: problems/789_escape_the_ghosts/solution.py
from typing import List


class Solution:
    def escapeGhosts(self, ghosts: List[List[int]], target: List[int]) -> bool:
        my_dist = abs(target[0]) + abs(target[1])
        for gx, gy in ghosts:
            if abs(gx - target[0]) + abs(gy - target[1]) <= my_dist:
                return False
        return True

File: problems/789_escape_the_ghosts/Solution.java
class Solution {
    public boolean escapeGhosts(int[][] ghosts, int[] target) {
        int myDist = Math.abs(target[0]) + Math.abs(target[1]);
        for (int[] g : ghosts)
            if (Math.abs(g[0] - target[0]) + Math.abs(g[1] - target[1]) <= myDist) return false;
        return true;
    }
}
